src/CommandLineParser.ts

```diff
@@ -0,0 +1,185 @@
+import { CLI_ERROR } from './Errors';
+
+import yargs from 'yargs';
+import colors from 'colors';
+import fs from 'fs';
+
+export const argv = yargs.options({
+    videoUrls: {
+        alias: 'V',
+        describe: 'List of video urls',
+        type: 'array',
+        demandOption: false
+    },
+    videoUrlsFile: {
+        alias: 'F',
+        describe: 'Path to txt file containing the urls',
+        type: 'string',
+        demandOption: false
+    },
+    username: {
+        alias: 'u',
+        type: 'string',
+        demandOption: false
+    },
+    outputDirectory: {
+        alias: 'o',
+        describe: 'The directory where destreamer will save your downloads [default: videos]',
+        type: 'string',
+        demandOption: false
+    },
+    outputDirectories: {
```

Comment:
I appreciate all the effort that went into implementing this, but why do we need this option? Why one directory per video?

Reply:
i need to download lots of videos and it is easier to save them in the right folder automatically, so that you don't need to open each video, because here all videos have the same name
it could be useful for someone else too though
one per video to make things simple

Reply:
I'm still not convinced but if it's under its own argument and if it helps you then i'll allow it :)